### PR TITLE
added tanh.md documentation

### LINKED ISSUE #3084: [Term Entry] Go Math `Tanh()`
### Reason/inspiration (optional)

We would like a new term entry for [Go](https://www.codecademy.com/resources/docs/go) about the `Tanh()` function.

The entry should go in a new file in `docs/content/go/concepts/math-functions/terms/tanh/tanh.md`

The entry should 

- Have a syntax section illustrating the function with a ` ```pseudo ` code block. 
- An example showing the function in use with a ` ```go ` code block. 
- A runnable codebyte example in a ` ```codebyte/golang` block. 

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by Dilandds**
### Description

Closes #3084 

Added the new file in docs/content/go/concepts/math-functions/terms/tanh/tanh.md. Which describes the functionality of Go language, Tanh() function in the Math library.

- Have a syntax section illustrating the function with a ```pseudo code block.
- An example showing the function in use with a ```go code block.
- A runnable codebyte example in a  ```codebyte/golang block.

### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)

### Checklist

- [ x] All writings are my own.
- [x ] My entry follows the Codecademy Docs style guide.
- [ x] My changes generate no new warnings.
- [x ] I have performed a self-review of my own writing and code.
- [x ] I have checked my entry and corrected any misspellings.
- [x ] I have made corresponding changes to the documentation if needed.
- [x ] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x ] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by SSwiniarski** [comment]
@Dilandds You are not supposed to submit a PR for an issue you haven't been assigned. Fortunately no one else had claimed the issue otherwise we'd have to close this PR. Please do not do so in the future.

File: content/go/concepts/math-functions/terms/tanh/tanh.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+Tags:
+  - 'Numbers'
+  - 'Hyperbolic Functions'
+  - 'Mathematics'
+CatalogContent:
+  - 'learn-go'
+  - 'paths/computer-science'
+---
+
+The **`Tanh()`** function returns the hyperbolic tangent of a given number. The `math` library must be imported in order to use this function.
```
Comment (yangc95) [inline content/go/concepts/math-functions/terms/tanh/tanh.md:16]:
```suggestion
The **`Tanh()`** function returns the hyperbolic tangent of a given number.
```

File: content/go/concepts/math-functions/terms/tanh/tanh.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+```
+
+`Tanh()` accepts an argument of type `float64` and returns a value of type `float64`.
+
+Note these special cases:
+
+-The result of `Tanh(math.Inf(-1))` is `-1`
+-The result of `Tanh(math.Inf(+1))` is `+1`
+-The result of `Tanh(0)` is `0`
+-The result of `Tanh(NaN)` is `NaN`
```
Comment (yangc95) [inline content/go/concepts/math-functions/terms/tanh/tanh.md:33]:
```suggestion
- The result of `Tanh(math.Inf(-1))` is `-1`
- The result of `Tanh(math.Inf(+1))` is `+1`
- The result of `Tanh(0)` is `0`
- The result of `Tanh(NaN)` is `NaN`
```

File: content/go/concepts/math-functions/terms/tanh/tanh.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+```pseudo
+import "math"
+
+result := math.Tanh(number)
+```
+
+`Tanh()` accepts an argument of type `float64` and returns a value of type `float64`.
+
+Note these special cases:
```
Comment (yangc95) [inline content/go/concepts/math-functions/terms/tanh/tanh.md:28]:
```suggestion
Special cases:
```

File: content/go/concepts/math-functions/terms/tanh/tanh.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+## Syntax
+
+```pseudo
+import "math"
+
+result := math.Tanh(number)
+```
+
+`Tanh()` accepts an argument of type `float64` and returns a value of type `float64`.
```
Comment (yangc95) [inline content/go/concepts/math-functions/terms/tanh/tanh.md:26]:
```suggestion
`Tanh()` accepts an argument, `number`,  of type `float64` and returns a value of type `float64`.
```

File: content/go/concepts/math-functions/terms/tanh/tanh.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+Note these special cases:
+
+-The result of `Tanh(math.Inf(-1))` is `-1`
+-The result of `Tanh(math.Inf(+1))` is `+1`
+-The result of `Tanh(0)` is `0`
+-The result of `Tanh(NaN)` is `NaN`
+
+`math.Inf(+1)` represents positive infinity, which is  `+Inf`.
+`math.Inf(-1)` represents negative infinity, which is  `-Inf`.
```
Comment (yangc95) [inline content/go/concepts/math-functions/terms/tanh/tanh.md:36]: are these special cases? if not, can you preface them

File: content/go/concepts/math-functions/terms/tanh/tanh.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+## Example
+
+The following calculates the hyperbolic tangent of 0.5 and prints out the result:
```
Comment (yangc95) [inline content/go/concepts/math-functions/terms/tanh/tanh.md:40]:
```suggestion
The following calculates the hyperbolic tangent of `0.5` and prints out the result:
```

File: content/go/concepts/math-functions/terms/tanh/tanh.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+  "math"
+)
+
+func main() {
+  value := math.Tanh(0.5)
+  fmt.Printf("%.2f\n", value)
+}
+```
+
+The output will be:
```
Comment (yangc95) [inline content/go/concepts/math-functions/terms/tanh/tanh.md:56]:
```suggestion
The output will be:

```

File: content/go/concepts/math-functions/terms/tanh/tanh.md (hunk 3 above)
Comment (yangc95) [reply content/go/concepts/math-functions/terms/tanh/tanh.md:28]: can you include why these are special cases

File: content/go/concepts/math-functions/terms/tanh/tanh.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+```shell
+0.46
+```
+
+## Codebyte 
+
+The following example is runnable and shows the Tanh() function handling the value 1.
```
Comment (yangc95) [inline content/go/concepts/math-functions/terms/tanh/tanh.md:63]:
```suggestion
The following example is runnable and shows the `Tanh()` function handling the value `1`.
```

Did you want to try one of the special cases here?

**Review by yangc95 — COMMENTED** [review]
@Dilandds conducted a review. Pleas let me know if you have any concerns!

File: content/go/concepts/math-functions/terms/tanh/tanh.md [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+
+Special cases:
+
+- The result of `Tanh(math.Inf(-1))` is `-1`
+- The result of `Tanh(math.Inf(+1))` is `+1`
+- The result of `Tanh(0)` is `0`
+- The result of `Tanh(NaN)` is `NaN`
+
+`math.Inf(+1)` represents positive infinity, which is  `+Inf`.
+`math.Inf(-1)` represents negative infinity, which is  `-Inf`.
```
Comment (yangc95) [inline content/go/concepts/math-functions/terms/tanh/tanh.md:36]:
```suggestion
- The result of `Tanh(math.Inf(-1))` is `-1`. `math.Inf(-1)` represents negative infinity, which is  `-Inf`.
- The result of `Tanh(math.Inf(+1))` is `+1`. `math.Inf(+1)` represents positive infinity, which is  `+Inf`.
- The result of `Tanh(0)` is `0`.
- The result of `Tanh(NaN)` is `NaN`.
```

**Review by yangc95 — APPROVED** [review]

**Pull request merged**